packages/wrangler/src/deployment-bundle/esbuild-plugins/hybrid-nodejs-compat.ts

```diff
@@ -0,0 +1,191 @@
+				getBasePath(),
+				`_virtual_unenv_global_polyfill-${globalName}.js`
+			)
+		),
+	];
+
+	build.onResolve({ filter: UNENV_GLOBALS_RE }, ({ path }) => ({ path }));
+
+	build.onLoad({ filter: UNENV_GLOBALS_RE }, ({ path }) => {
+		// eslint-disable-next-line  @typescript-eslint/no-non-null-assertion
+		const globalName = path.match(UNENV_GLOBALS_RE)![1];
+		const globalMapping = inject[globalName];
+
+		if (typeof globalMapping === "string") {
+			const globalPolyfillSpecifier = globalMapping;
+
+			return {
+				contents: `
+								import globalVar from "${globalPolyfillSpecifier}";
+
+								${
+									/*
+								// ESBuild's inject doesn't actually touch globalThis, so let's do it ourselves
+								// by creating an exportable so that we can preserve the globalThis assignment if
+								// the ${globalName} was found in the app, or tree-shake it, if it wasn't
+								// see https://esbuild.github.io/api/#inject
+								*/ ""
+								}
+								const exportable =
+									${
+										/*
+									// mark this as a PURE call so it can be ignored and tree-shaken by ESBuild,
+									// when we don't detect 'process', 'global.process', or 'globalThis.process'
+									// in the app code
+									// see https://esbuild.github.io/api/#tree-shaking-and-side-effects
+									*/ ""
+									}
+									/* @__PURE__ */ (() => {
+										${
+											/*
+										// TODO: should we try to preserve globalThis.${globalName} if it exists?
+										*/ ""
+										}
+										return globalThis.${globalName} = globalVar;
+									})();
+
+								export {
+									exportable as '${globalName}',
+									exportable as 'globalThis.${globalName}',
+								}
+							`,
+			};
+		}
+
+		const [moduleName, exportName] = inject[globalName];
+
+		return {
+			contents: `
+							import { ${exportName} } from "${moduleName}";
+
+							${
+								/*
+							// ESBuild's inject doesn't actually touch globalThis, so let's do it ourselves
+							// by creating an exportable so that we can preserve the globalThis assignment if
+							// the ${globalName} was found in the app, or tree-shake it, if it wasn't
+							// see https://esbuild.github.io/api/#inject
+							*/ ""
+							}
+							const exportable =
+								${
+									/*
+								// mark this as a PURE call so it can be ignored and tree-shaken by ESBuild,
+								// when we don't detect 'process', 'global.process', or 'globalThis.process'
+								// in the app code
+								// see https://esbuild.github.io/api/#tree-shaking-and-side-effects
+								*/ ""
+								}
+								/* @__PURE__ */ (() => {
+									${
+										/*
+									// TODO: should we try to preserve globalThis.${globalName} if it exists?
+									*/ ""
+									}
+									return globalThis.${globalName} = ${exportName};
+							})();
+
+							export {
+								exportable as '${globalName}',
+								exportable as 'global.${globalName}',
+								exportable as 'globalThis.${globalName}'
```

Comment:
@petebacondarwin can you check these too in the fixture test you proposed?

Reply:
Done in 0ec2e8632ced68a2db402223a73f84b893643be7